service/mantle/shipstation/ShipStationServices.xml

```diff
@@ -0,0 +1,116 @@
+                <description>Find orders placed at or before this date.</description>
+            </parameter>
+            <parameter name="page"/>
+        </in-parameters>
+        <actions>
+
+            <set field="orderParts" from="[]"/>
+            <script>
+
+                def response = ec.web.getResponse();
+                def context =
+                ec.service.sync().name("co.hotwax.oms.OrderServices.find#Orders").parameters([statusId:'OrderApproved',placedDateStart:ec.l10n.format(start_date,'yyyy-MM-dd
+                HH:mm'),placedDateEnd:ec.l10n.format(end_date,'yyyy-MM-dd HH:mm')]).call()
+
+                orderParts = context.orders
+                response.setContentType("application/xml")
+                Writer w = response.getWriter()
+                ec.resource.template("component://mantle-shipstation/screen/OrderData.xml.ftl",w)
+                w.flush()
+                w.close()
+            </script>
+
+        </actions>
+    </service>
+
+    <!-- service for POST call, when ShipStation wants to POST Shipment Notification-->
+    <service verb="post" noun="ShipNotification">
+        <in-parameters>
+            <parameter name="order_number"/>
+            <parameter name="carrier"/>
+            <parameter name="service"/>
+            <parameter name="tracking_number"/>
+        </in-parameters>
+        <actions>
+            <set field="xmlData" from="ec.web.getRequestBodyText()"/>
+            <set field="itemList" from="[]"/>
+            <script>
+
+                def rootNode = new XmlParser().parseText(xmlData)
+                def orderId = rootNode.'OrderID'.text()
+                def shipDate = rootNode.'ShipDate'.text()
+                def Shipping_date = Date.parse("MM/dd/yyyy", shipDate).format("yyyy-MM-dd")
+                //def ShippingCost = rootNode.'ShippingCost'.text()
+                def Carrier = rootNode.'Carrier'.text()
+                def trackingNo = rootNode.'TrackingNumber'.text()
+                def labelDate = rootNode.'LabelCreateDate'.text()
+                def labelCreateDate = Date.parse('MM/dd/yyyy HH:mm', labelDate).format("yyyy-MM-dd HH:mm")
+
+                for (Node item in rootNode.'Items'.'Item') {
+                String sku = item.'SKU'.text()
+                String name = item.'Name'.text()
+                String quantity = item.'Quantity'.text()
+                String lineItemId = item.'LineItemID'.text()
+                itemList.add([SKU:sku,Name:name,quantity:quantity,LineItemId:lineItemId])
+                }
+            </script>
+
+            <entity-find-one entity-name="mantle.order.OrderHeaderAndPart" value-field="orderInfo">
+                <field-map field-name="orderId" from="orderId"/>
+            </entity-find-one>
+
+            <service-call name="mantle.shipment.ShipmentServices.create#Shipment" in-map="[shipmentTypeEnumId:'ShpTpSales',statusId:'ShipInput',fromPartyId:orderInfo.vendorPartyId,
```

Comment:
Can you please add some comments on how shipment information is being saved on each service call. The code must have proper comments.